buildsystem/Doxygen.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014-2015 the openage authors. See copying.md for legal info.
+
+option(GENERATE_DOCS "Generate documentation from the source files." ON)
+
+set(DOXYFILE_TEMPLATE "${BUILDSYSTEM_DIR}/templates/Doxyfile.in")
+set(DOCS_ROOT_DIRECTORY "${CMAKE_BINARY_DIR}/docs")
+
+function(add_doxygen_target TARGET TARGET_SRC_DIR)
+	if(NOT DOXYGEN_FOUND)
+		return()
+	endif()
+
+	set(DOXYFILE "${CMAKE_BINARY_DIR}/${TARGET}_Doxyfile")
+	set(OUTPUT_DIRECTORY "${DOCS_ROOT_DIRECTORY}/${TARGET}")
```

Comment:
It's not that i want them in a single folder, I want them in the same doxygen output.
Maybe (I'm not sure) it is better to have totally separate doxygen invocations, but when we combine them it may lead to name clashes between c++ and python names.

What do you think? Should the same webpage include docs for both py and c++?